### DISCUSSION TITLE
Simple Zoom Room Quick Links

### DISCUSSION THREAD

**Discussion by peterjohnhunt** [johnlindquist/kit#138] [Share]
Hey @johnlindquist! First off, thank you so much for this app! one of the coolest things i've played around with in a long time! I've already spent a bunch of time prototyping a bunch of scripts and automations. Definitely is going to make much easier a lot of the automation type scripts i usually end up writing in bash a bit more clunkily.

I built a super simple zoom room quick launcher: https://gist.github.com/peterjohnhunt/043b9051ff097d415efc8215889d6a25

**A couple questions:**

1) For me when doing certain actions (like trying to share) via the UI, it seems to "get confused". For instance, when i try and share a script via "Create install link" it asks me for a github token, but when i enter it i get this error:

`[warn] ENOENT: no such file or directory, open '/Users/peterjohnhunt/.kenv/scripts/MYTOKENHERE`

It feels like it's using the token as the file it's supposed to share. Doing the share process through the terminal works fine, and after the token is set, the UI works fine as well (since it doesn't prompt for a token)

2) Following the pattern of the "todos app" i built a db interface as a "lib". Is there an easy way to share the full "script" including that file to others?

3) For something like opening a url in the default browser, is exec("open MYURL") the best method, or is there a helper function i'm missing?

4) One other thought, it'd be cool if there was a programatic way to change the tab. An example scenario: After adding a new item to the db, it would be cool if i could trigger it to auto change back to the main tab.

**Comment by johnlindquist** [comment]
Oh nice, a Zoom script! That's gonna be super useful. Thanks so much for sharing.

1. I imagine the `env` internals are a bit outdated. I've refactored a lot of things without touching the env logic in a while. I need to revisit it soon.
2. Not yet. I'm hoping to add logic that parses the script, finds the `lib` line, and pulls in all the libs to share, but that's still a fairly low priority, unfortunately.
3. `exec` is the "best' so far. I think a lot about an `open` helper, but I haven't really settled on anything. 
4. The current "api" for tab-switching is:

```js
await run("todos", "--tab", "Toggle")
```
That will run the todos script focusing on the "ToggleI tab. I've been trying to keep the terminal usage into consideration which would be:
```bash
todos --tab Toggle
```

If I ever write a tab switch helper, it would probably just wrap that code

Wonderful questions. Glad to see you diving in. I hope to be able to sort out all the beta hiccups over the coming weeks.
